src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -47,34 +47,37 @@ protected virtual HttpClient GetHttpClient()
             IDictionary<string, string> headers,
             IDictionary<string, string> bodyParameters,
             ILoggerAdapter logger,
+            bool useManagedIdentity = false,
             CancellationToken cancellationToken = default)
         {
             HttpContent body = bodyParameters == null ? null : new FormUrlEncodedContent(bodyParameters);
-            return await SendPostAsync(endpoint, headers, body, logger, cancellationToken).ConfigureAwait(false);
+            return await SendPostAsync(endpoint, headers, body, logger, useManagedIdentity: useManagedIdentity, cancellationToken).ConfigureAwait(false);
```

Comment:
This is violating the single responsability principle, i.e. HttpManager is only responsible for making Http calls, it should not know anything about MSI.